udata/core/spatial/models.py

```diff
@@ -101,6 +101,11 @@ def url(self):
     def external_url(self):
         return endpoint_for('territories.territory', territory=self, _external=True)
 
+    def count_datasets(self):
+        from udata.models import Dataset
+        self.metrics['datasets'] = Dataset.objects(spatial__zones__contains=self.id).visible().count()
```

Comment:
Why the contains? This job seems to take a lot of time (~1h).

Reply:
`__contains` is a string operator I think. You can just do `spatial_zones=` to [query a list element](https://docs.mongoengine.org/guide/querying.html#querying-lists).

Reply:
Hum… Maybe it'll do a `contains` on all element of the list to check… It could explain the job long duration time…